app/models/nic/managed.rb

```diff
@@ -128,7 +128,7 @@ def failure(msg, backtrace = nil, dest = :base)
 
     # we must also clone host object so we can detect host attributes changes
     def setup_clone
-      return if new_record?
+      return if new_record? or host.new_record?
```

Comment:
This was the cause of the failing test in host_test - it seems it's possible to have Host as new_record when Nic is not. I don't *think* it makes sense to run clones in this scenario, but could you confirm @ares ?